python-threatexchange/threatexchange/hashing/pdq_faiss_matcher.py

```diff
@@ -209,9 +255,23 @@ def mih_index(self):
             return faiss.downcast_IndexBinary(self.faiss_index.index)
         return self.faiss_index
 
-    def search(self, queries: t.Sequence[PDQ_HASH_TYPE], threshhold: int, **kwargs):
+    def search(
+        self,
+        queries: t.Sequence[PDQ_HASH_TYPE],
+        threshhold: int,
+        return_as_ids: bool = False,
+    ):
+        self.mih_index.nflip = threshhold // self.mih_index.nhash
+        return super().search(queries, threshhold, return_as_ids)
+
+    def search_with_distance(
```

Comment:
blocking: Wait a second, doesn't this shadow your two definitions on L51 and L104?

Reply:
it overrides then call the base version. because for `PDQMultiHashIndex` there are so prep work before a query based on threshold